Add Documentation for R4 Encounter Write

JFisk42 opened the pull request:
Not to be merged until after https://github.com/cerner/fhir.cerner.com/pull/238 is in master.

JFisk42 commented on lib/resources/r4/types.yaml:
```diff
@@ -0,0 +1,30 @@
+---
```
So this file is required by https://github.com/cerner/fhir.cerner.com/blob/master/lib/definition_table_generator.rb#L33 to generate a yaml table. Not sure what exactly a minimum set might look like but I'll play around with it some.

daliboz replied:
There's a bit of info about how this works/what it is for on the readme: https://github.com/cerner/fhir.cerner.com

JFisk42 replied:
I see thanks! Is there a preference on which to use for the url? either the resource specific link like http://hl7.org/fhir/r4/encounter-definitions.html#Encounter.identifier or the field dataType link http://hl7.org/fhir/r4/datatypes.html#Identifier?

MaxPhilips replied:
assuming datatypes since that's what DSTU2 used

JFisk42 replied:
Added a minimum set of datatype urls to r4/types.yaml and removed the specific urls from the encounter yaml: https://github.com/cerner/fhir.cerner.com/pull/239/commits/512b564a5cb8f711b3cb4e115e8ac24b45f62fcb

JFisk42 commented on lib/resources/r4/encounter.yaml:
```diff
@@ -0,0 +1,236 @@
+  required: 'Yes'
+  cardinality: 1..1
+  type: Reference (Patient)
+  description: The patient present at the encounter.
```
Needs to add "action: create"

JFisk42 replied:
added: https://github.com/cerner/fhir.cerner.com/pull/239/commits/3bbb3b5c2de845dc9a6494d942a5ee43f7ef9198

JFisk42 commented on content/millennium/r4/encounters/encounter.md:
```diff
@@ -97,3 +97,56 @@ The common [errors] and [OperationOutcomes] may be returned.
+
+### Headers
+
+<%= headers head: {Authorization: '&lt;OAuth2 Bearer Token>', 'Content-Type': 'application/json+fhir'} %>
```
Should be application/fhir+json for content-type

JFisk42 replied:
added: https://github.com/cerner/fhir.cerner.com/pull/239/commits/b320b5dc6baee634a654bc534bd79229818fa6fe

MaxPhilips replied:
Eddy added this to the headers method https://github.com/cerner/fhir.cerner.com/commit/761f781e1eb58169b9b02d2530719b468b68a24d#diff-7718c11ed130ac00a9854e0b089f74b4L27

JFisk42 replied:
It looks like that doesn't apply here because default_headers_r4 only pulls in accept type and not content type.

JFisk42 commented on content/millennium/r4/encounters/encounter.md:
```diff
@@ -97,3 +97,56 @@ The common [errors] and [OperationOutcomes] may be returned.
+
+### Errors
+
+The common [errors] and [OperationOutcomes] may be returned.
```
This will need to be updated when an R4 OperationOutcome page is made.

JFisk42 commented on content/millennium/r4/encounters/encounter.md:
```diff
@@ -97,3 +97,56 @@ The common [errors] and [OperationOutcomes] may be returned.
+
+### Authorization Types
+
+<%= authorization_types(practitioner: true, system: true) %>
```
add "patient: false"

JFisk42 replied:
added: https://github.com/cerner/fhir.cerner.com/pull/239/commits/b320b5dc6baee634a654bc534bd79229818fa6fe

MaxPhilips commented on content/millennium/r4/encounters/encounter.md:
```diff
@@ -97,3 +97,56 @@ The common [errors] and [OperationOutcomes] may be returned.
+
+### Body Fields
+
+<%= definition_table(:encounter, :create, :r4) %>
```
this table scrolls... any way we can only scroll the code sections instead of the whole thing

JFisk42 replied:
I'm not sure if we have any fine control over this? Patient works this way too: https://fhir.cerner.com/millennium/dstu2/individuals/patient/#body-fields

MaxPhilips commented on content/millennium/r4/encounters/encounter.md:
```diff
@@ -97,3 +97,56 @@ The common [errors] and [OperationOutcomes] may be returned.
+
+<%= headers status: 201 %>
+<pre class="terminal">
+    Cache-Control → no-cache
```
let's use colons instead of weird arrow characters

JFisk42 replied:
changed: https://github.com/cerner/fhir.cerner.com/pull/239/commits/9c48980f8f6217f0ba08e6a136317e557903f83e

MaxPhilips replied:
can we remove the leading spaces too?

JFisk42 replied:
removed: https://github.com/cerner/fhir.cerner.com/pull/239/commits/151548d450d95be176f9e63386de7cbd34a6e28a

MaxPhilips commented on content/millennium/r4/encounters/encounter.md:
```diff
@@ -97,3 +97,56 @@ The common [errors] and [OperationOutcomes] may be returned.
+    Server-Response-Time → 3890.363996
+    Status → 201 Created
+    Vary → Origin
+    X-Newrelic-App-Data → PxQGVl9WCQcBR1BaBwMHU1wCBBFORCANXhZKDVRUUUAcA1EIQz5WDAdZQg8SBkpsVQoGC1lUGAMKUllGDBIGEG8BSgdZTFEdUBdcAEUEEU5UGAdNVU0ACgBTU1sDAwBVVwYAB1RTT08BThoGD14BVwVSDAQFBwVSAQ5TQzs=
```
remove

JFisk42 replied:
removed: https://github.com/cerner/fhir.cerner.com/pull/239/commits/9c48980f8f6217f0ba08e6a136317e557903f83e

krabey commented on lib/resources/r4/encounter.yaml:
```diff
@@ -0,0 +1,399 @@
+---
```
looks like a few proprietary codeset examples are missing:
identifier.type: codeSet/319
hospitalization.dietPreference: codeSet/18
hospitalization.specialCourtesy: codeSet/67
hospitalization.specialCourtesy: codeSet/16

JFisk42 replied:
Added an example for identifier.type: https://github.com/cerner/fhir.cerner.com/pull/239/commits/40b3c670db01bc6b48dbb783e357036bee956549

Encounter doesn't write the other hospitalization fields.

JFisk42 commented:
I'll have to push my rebase commit later tonight. I rebased on the newest read/search changes and the fields are overwriting the write fields on the body fields section.

JFisk42 commented:
Added some identifier subfields, adjusted a few things so the write body fields didn't pull in extra fields. https://github.com/cerner/fhir.cerner.com/pull/239/commits/fefea58e204f031382e294344ba4c49fe25a66a9
